pinot-query-planner/src/main/java/org/apache/pinot/query/timeboundary/TimeBoundaryStrategyFactory.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.timeboundary;
+
+import org.apache.pinot.spi.data.TimeBoundaryConfig;
+
+
+public class TimeBoundaryStrategyFactory {
```

Comment:
Can you take a look at https://github.com/apache/pinot/blob/master/pinot-spi/src/main/java/org/apache/pinot/spi/cursors/ResponseStoreService.java ? Right now, this is the preferred way to add plugins. Cursors, indexes (IndexPlugin) all use this strategy. It avoids a switch case and loads all the implementations of a certain type. This will make it easier to add plugins in forks as well without having to change code in OSS.